### ISSUE TITLE
Remove `IPython.utils.warn`

### ISSUE THREAD

**Issue by Carreau** [ipython/ipython#9248]
cf #9090, it's been deprecated, it can now be removed.

**Comment by minrk** [comment]
Where by "now" we mean after we've released 5.0 at some time in the future.

**Comment by Carreau** [comment]
yes, yes, so much nitpicking :-)

**Comment by minrk** [comment]
I just don't want to give the wrong impression that opening a PR for this now would be a good idea.

**Comment by Carreau** [comment]
Adding a bit more description to this issue to be clearer. 

The code which is present in the file `IPython/utils/warn.py` was used for a long time by IPython to emit warnings; during the last few years and month, we replaced the usage of `/IPython/utils/warn.py` with the builtin Python warn module.

It seem like the `IPython/utils/warn.py` module is not use anymore, though we left it here for 1 release in order to give our users one last chance to update to the new API if needed without raising an `ImportError` exception

Now that we are starting to work on IPython 6.0 and since this module is not used, it can likely just be removed. 

This mean that the file `IPython/utils/warn.py` can be deleted, and likely we can add a file in `docs/source/whatsnew/` that says something along the line of  `IPython.utils.warn was deprecated in IPtyhon 4.0, and has now been removed`. 

This should be a relatively straightforward issue for a new contributor that look to contribute to open source and looks to start helping. 

If you are a user who is looking to take care of this issue and have any questions, feel free to ask; and or indicate you interest in working on this.
